app/controllers/admin/column_preferences_controller.rb

```diff
@@ -10,12 +10,17 @@ def bulk_update
       if @cp_set.save
         respond_to do |format|
           format.json { render json: @cp_set.collection, each_serializer: Api::Admin::ColumnPreferenceSerializer }
-          format.html { render inline: "saved" } #todo
+          format.turbo_stream {
+            flash.now[:success] = t('.success')
+            render :bulk_update, locals: { action: permitted_params[:action_name] } }
         end
       elsif @cp_set.errors.present?
         respond_to do |format|
           format.json { render json: { errors: @cp_set.errors }, status: :bad_request }
-          format.html { render inline: "errors" } #todo
+          format.turbo_stream {
+            flash.now[:error] = t('.error')
+            render :bulk_update, locals: { action: permitted_params[:action_name] }
+          }
```

Comment:
Just clarifying: we are using a simple HTTP response for turbo streams, right? Not websockets?

Reply:
Yep, I just double-checked to be 100% sure.

![Screenshot 2024-06-13 at 5 02 16 pm](https://github.com/openfoodfoundation/openfoodnetwork/assets/4188088/479858bc-b6cd-46f3-a279-ae6091dd2a88)